janus/message-base.md

```diff
@@ -0,0 +1,306 @@
+# SSI Base Message Format
+
+> *Note: This is a rough document, and contains inconsistancies, missing sections, etc. These will iron out as work on this document progresses.*
+
+The role of the Self-Soverign Identity Message Format is to act as a foundation for the message types that will power the SSI messaging ecosystem.
+This format is intended to be as simple as possible, providing just enough structure to facilitate compatible communication.
+
+Goals:
+* Broad compatibility across SSI ecosystem
+* Allow for innovation in message types
+* Support communication from clients with commonly available libraries
+
+## Base Envelope
+
+The base message format is a JWT Payload. All JWT standard attributes are reserved. Additional attributes to messages are included based on the type specified. (See Message Types.)
+
+Example:
+```JSON
+{
+    "@type": "example.com/basemessage",
+    "iss": "did:exa:11111111111",
+    "aud": "did:exa:22222222222",
+    "iat": "timestamp",
+}
+```
+### Applicable JWT attributes
+
+* **"iss"** (Issuer) - from DID
+* **"sub"** (Subject)
+* **"aud"** (Audience) - to DID
+* **"exp"** (Expiration Time) - ts
+* **"nbf"** (Not Before) - ts
+* **"iat"** (Issued At) - ts
+* **"jti"** (JWT ID) - unique id for JWT to prevent replay
+
+Additional JOSE Headers (for JWS/JWE):
+
+* **"typ"** (Type) - Should these be used as type designators?
+* **"cty"** (Content Type)
+
+
+## Message Types
+
+Each message should have a type, specified by the top level @type attribute. The attribute value should be a URI that uniquely identifies the message type, and should be resolvable into a resource (ie, webpage) with details for message type use.
+
+Each type indicates a particular version and format. 
+
+```JSON
+{
+  "@type": "example.com/examplemessage/v1"
```

Comment:
This should be flexible enough to reference a web address as well as a blockchain schema address. Perhaps this should be a URI with a couple of examples, one web, and one blockchain?